main/main.ino

```diff
@@ -30,14 +30,17 @@
 // array to store previous received RFs, IRs codes and their timestamps
 #if defined(ESP8266) || defined(ESP32) || defined(__AVR_ATmega2560__) || defined(__AVR_ATmega1280__)
 #  define array_size 12
-unsigned long ReceivedSignal[array_size][2] = {{0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}};
+unsigned long long ReceivedSignal[array_size][2] = {{0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}};
```

Comment:
Due to the fact that there is already a condition on boards here, I can use it anyway if you think it is better

Reply:
This also relates to the general question "How do we decided where to use the TYPE_UL_ULL type?". As I see it, we introduced this type macro for the handling of 64 bit codes in the RF and IR modules. This is fine, but then we should only use this macro for such codes and use this macro everywhere we handle such codes.

The `storeValue` and `isAduplicate` functions are in the `main.ino` and therefore provided as a feature to all modules.  These functionality is strongly related to the two modules and I think should not be in the main.ino. Else they should not expose the type of the RF and IR codes. The functions should at least document that they are related to RF and IR dedup and should not be used otherwise, then I think it's fine to use the macro here. I then would also rename the macro to make it clear it represent the size of the RF and IR codes.

The type of `ReceivedSignal` should be changed, it represents the codes in the first array and the times in the second, but the times are only `unsigned long` as you can see in the `isAduplicate` function. So the 2d array is not the best data structure to represent this type of data.

Reply:
Thanks for the comments.

> This is fine, but then we should only use this macro for such codes.

`value` key is used by IR, RF and SRFB gateways
`storeValue` and `isAduplicate` functions are used in `ZgatewayWeatherStation` with 32 bits value, `ZgatewaySRFB` with 32 bits value, `ZgatewayIR` with 32 or 64 bits and `ZgatewayRF` with 32 or 64 bits 

I took the hypothesis that a 64bits variable can handle a 32bits ones for `ZgatewaySRFB` .
So the functions in `main.ino`  should be able to handle the 32 bits values.

Maybe the only issue is here:
`if (p.value.is<TYPE_UL_ULL>()`

But we are in `simplePublishing `scope, so will be removed soon and not a lot used.

>The storeValue and isAduplicate functions are in the main.ino and therefore provided as a feature to all modules. These functionality is strongly related to the two modules and I think should not be in the main.ino. Else they should not expose the type of the RF and IR codes. The functions should at least document that they are related to RF and IR dedup and should not be used otherwise, then I think it's fine to use the macro here. I then would also rename the macro to make it clear it represent the size of the RF and IR codes.

ok, I will modify the code

> The type of ReceivedSignal should be changed, it represents the codes in the first array and the times in the second, but the times are only unsigned long as you can see in the isAduplicate function. So the 2d array is not the best data structure to represent this type of data.

ok, I will change the data structure